rest/api.go

```diff
@@ -159,6 +159,8 @@ type API interface {
 	// responseSerializer returns a ResponseSerializer for the given format type. If the
 	// format is not implemented, the returned serializer will be nil and the error set.
 	responseSerializer(string) (ResponseSerializer, error)
+
+	Router() *mux.Router
```

Comment:
I don't like the idea of exposing `mux` on the `API` interface. This interface is meant to be generic. `mux` is just one specific backing implementation. I would prefer to avoid exposing this altogether, but if we must, make it an interface or something which `mux` can implement.